src/snapred/backend/data/LocalDataService.py

```diff
@@ -119,6 +168,68 @@ def _readInstrumentParameters(self) -> Dict[str, Any]:
         except FileNotFoundError as e:
             raise _createFileNotFoundError("Instrument configuration file", self.instrumentConfigPath) from e
 
+    def _findMatchingFileList(self, pattern, throws=True) -> List[str]:
+        """
+        Find all files matching a glob pattern.
+        Optional: throws exception if nothing found.
+        """
+        fileList: List[str] = []
+        for fname in glob.glob(pattern, recursive=True):
+            if os.path.isfile(fname):
+                fileList.append(fname)
+        if len(fileList) == 0 and throws:
+            raise ValueError(f"No files could be found with pattern: {pattern}")
+
+        return fileList
+
+    def _findMatchingDirList(self, pattern, throws=True) -> List[str]:
+        """
+        Find all directories matching a glob pattern.
+        Optional: throws exception if nothing found.
+        """
+        fileList: List[str] = []
+        for fname in glob.glob(pattern, recursive=True):
+            if os.path.isdir(fname):
+                fileList.append(fname)
+        if len(fileList) == 0 and throws:
+            raise ValueError(f"No directories could be found with pattern: {pattern}")
+
+        return fileList
+
+    def getIPTS(self, runNumber: str, instrumentName: str = Config["instrument.name"]) -> str:
+        key = (runNumber, instrumentName)
+        if key not in self.iptsCache:
+            self.iptsCache[key] = GetIPTS(RunNumber=int(runNumber), Instrument=instrumentName)
+            # pull all run numbers from this IPTS folder, as they will likely also come up
+            if os.path.isdir(self.iptsCache[key]):
+                neutronDataFileGlob = self.iptsCache[key] + f"/{Config['instrument.name']}_*.nxs.h5"
+                for fname in glob.glob(neutronDataFileGlob, recursive=False):
+                    # extract out the run number
+                    res = re.search(self.iptsCache[key] + f"/{Config['instrument.name']}_" + "(\d+)\..*", fname)
+                    # if a run number was found
+                    if res is not None:
+                        self.iptsCache[(res.group(1), instrumentName)] = self.iptsCache[key]
```

Comment:
I had discussed this with Pete.  Since runs in the same IPTS are likely related and likely to come up again, it makes sense to pull them all in to avoid calling GetIPTS even more.